src/Adapters/EventSource.Net46/Implementation/EventDataExtensions.cs

```diff
@@ -0,0 +1,198 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="EventDataExtensions.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2017
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.DataContracts;
+
+namespace Microsoft.ApplicationInsights.EventSourceListener.Implementation
+{
+    internal static class EventDataExtensions
+    {
+        private static Lazy<Random> random = new Lazy<Random>();
+        private static SeverityLevel[] EventLevelToSeverityLevel = new SeverityLevel[]
+        {
+            SeverityLevel.Critical,     // EventLevel.LogAlways == 0
+            SeverityLevel.Critical,     // EventLevel.Critical == 1
+            SeverityLevel.Error,        // EventLevel.Error == 2
+            SeverityLevel.Warning,      // EventLevel.Warning == 3
+            SeverityLevel.Information,  // EventLevel.Informational == 4
+            SeverityLevel.Verbose       // EventLevel.Verbose == 5
+        };
+
+        /// <summary>
+        /// Creates a TraceTelemetry out of an EventSource event and tracks it using the supplied client.
+        /// </summary>
+        /// <param name="eventSourceEvent">The source for the telemetry data</param>
+        /// <param name="client">Client to track the data with</param>
+        public static void Track(this EventWrittenEventArgs eventSourceEvent, TelemetryClient client)
+        {
+            Debug.Assert(client != null);
+
+            string formattedMessage = null;
+            if (eventSourceEvent.Message != null)
+            {
+                try
+                {
+                    // If the event has a badly formatted manifest, message formatting might fail
+                    formattedMessage = string.Format(CultureInfo.InvariantCulture, eventSourceEvent.Message, eventSourceEvent.Payload.ToArray());
```

Comment:
If we do not format the message, it will show in AI as unformatted string. In extreme, but very popular case (a "Message" type event) the value will be "{0}". Very un-informative.